aips/aip-116.md

```diff
@@ -234,17 +241,50 @@ message-resources = \\*( LF "- " URI )
 
 ### Domain Separated Signing Algorithm
 
-A plain text construction of a SIWA message must be prefixed with a `SIGN_IN_WITH_APTOS::` hash before signing to ensure domain separation across other signatures. The SIWA signing messages must conform to the following expression:
+A plain text construction of a SIWA message (as described [here](#siwa-message-construction)) must be prefixed with a `SIGN_IN_WITH_APTOS::` hash before signing to ensure domain separation across other signatures. The SIWA signing messages must conform to the following expression:
 
 ```
 sha3_256(b"SIGN_IN_WITH_APTOS::") || <siwa_message>
 ```
 
 where `<siwa_message>` is the plain text construction of a SIWA message explained in the [Message Construction](#message-construction) section.
 
+### Signature and Message Verification
+
+The verification process consists of two equally important components:
+
+1. **Signature verification** - Cryptographically ensures the authenticity of a signature against the `message`.
+2. **SIWA Message verification** - Ensures the content of the `AptosSignInOutput` message adheres to the expected fields.
+
+#### Signature Verification
+
+Signature verification must consist of three parts:
+
+- `publicKey` - The public key of the user that is signing in.
+- `signature` - The signature to verify.
+- `message` - The SIWA message to verify.
+
+To verify the `signature`, the `publicKey` should be used to cryptographically verify the `signature` against the `message`. This verification should be stateless.
+
+#### Message Verification
+
+The message verification process must consist of the following parts:
+
+- `publicKey` - The public key of the user that is signing in. Should be the `publicKey` used in the `signature` verification step above.
+- `expected` - The `AptosSignInInput` fields that are expected to be present in the `message`.
```

Comment:
There is a problem here for me: I have no idea what should be in this `expected` fields.